src/irmin-pack/IO.ml

```diff
@@ -148,6 +166,7 @@ module Unix : S = struct
     Buffer.clear t.buf
 
   let v ~version ~fresh ~readonly file =
+    (* check the version is [Some _]; fail if not *)
```

Comment:
Personal preference, but I think the code makes this relatively clear here already.

Reply:
I feel the comment usefully summarises the code. Do you think it makes the code less understandable? But there is another reason for the comment.

In this case, when I started reading the code, I had no expectation for what "version" should mean, or how the code should behave wrt. the version. Thus, when I read that the code fails if there is a version mismatch, that is new information that I didn't have before. So I put in a comment to make clear that this is defining the semantics to some extent. Of course, I then try to make sure the doc in IO_intf matches the detail of the implementation.

Reply:
Personally I find the implementation of `get_version` pretty self-descriptive, so it seems valid to learn what it does by reading its implementation rather than a comment. My first instinct when logic becomes complex enough to require a comment is generally to refactor to include the information in-band instead. e.g. this function could become `get_version_for_fresh_store`, which helps clarify the call-sites as well as the definition site.

I don't feel strongly enough about this to split hairs over it though :-) As you wish.